contracts/token/WithdrawTokens.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC721/IERC721Upgradeable.sol";
+
+/**
+ * @title Making a token withdrawable by the owner to a specified onlyOwner setable address
+ *
+ * @dev
+ */
+abstract contract WithdrawTokens is OwnableUpgradeable {
+    address public beneficiary;
+
+    event BeneficiaryUpdated(address beneficiary);
+
+    constructor(address _owner) {
+        transferOwnership(_owner);
+    }
+
+    function setBeneficiary(address _beneficiary) public onlyOwner {
+        require(_beneficiary != address(0), "WithdrawTokens/beneficiary zero");
+        beneficiary = _beneficiary;
+        emit BeneficiaryUpdated(_beneficiary);
+    }
+
+    function withdrawERC20Tokens(IERC20Upgradeable token, uint256 amount)
+        public
+        onlyOwner
+    {
+        // can we safely make this non-onlyOwner?
```

Comment:
suggestion: input validation.

The address must be non-zero and not `address(this)`

Could also validate the token.

Reply:
> The address must be non-zero and not address(this)

I added these checks to `updateBeneficiary()` since thats the only way to set the `to` address?

Validate the token how? ERC165?

Reply:
No those checks are to ensure against a user error.

This will be used by the Bond contract (an ERC20) so you could transfer to itself ...which is a non-op, but you'd still end up paying gas for the privilege.

Those input guards would revert the transaction and save the user from that inconvenience.